### PR TITLE
migrate to androidx + other changes

### PR DESCRIPTION
**Pull request by TacoTheDank**
some other changes:
- reformatted code
- add gradle to project
- simplified a return
- migrated app to appCompat
- migrated to androidx

### CODE REVIEW

**Comment by TacoTheDank** [comment]
@adrian-bl If you could accept this, I can start working on the apps

**Comment by adrian-bl** [comment]
@Adonai is the developers of the plugins

**Comment by Kaned1as** [comment]
I can't merge this as-is. @TacoTheDank , you have to understand one thing. This repo is not a library in itself, it's a submodule of every plugin. I tried building Cover Fetch and it failed on compilation right away. Have you tried building any of the plugins using your branch?

image


I don't actually mind if you can convert it into a library gracefully, but please test your changes before you create a PR.

**Comment by TacoTheDank** [comment]
@adrian-bl ah ok

@Adonai That's because it's migrated to AndroidX, and the plugin apps are still using the support libraries. The issue occurs because you are using it as a submodule (correct me if I'm wrong tho). This would be much easier to deal with if it were a library.

To use this as a library, I'd suggest using "https://jitpack.io", as it does everything automatically, and is very easy to use. Sign into Jitpack using GitHub. There will be a bar of all your repositories on the left. Go to vanilla-music-plugin-commons. Basically this: (except yours will probably be ```implementation 'com.github.vanilla-music:vanilla-music-plugin-commons:Tag```
image

**Comment by TacoTheDank** [comment]
Undid the commits, added that bug fix commit I somehow missed, and redid the commits (so no unnecessary pull requests have to be made)

File: build.gradle [hunk 1]
```diff
@@ -15,12 +32,15 @@ android {
     }
 
     lintOptions {
-        disable 'MissingTranslation'
-        disable 'ExtraTranslation'
+        disable 'MissingTranslation', 'ExtraTranslation'
+        abortOnError false
```
Comment (Kaned1as) [inline build.gradle]: Why's that? Abort is abort.

File: src/main/res/values/styles.xml [hunk 2]
```diff
@@ -7,7 +7,7 @@
         <item name="android:windowIsFloating">false</item>
         <item name="android:windowContentOverlay">@null</item>
         <item name="android:windowSoftInputMode">stateAlwaysHidden</item>
-        <item name="android:windowActionModeOverlay">true</item>
+        <item name="windowActionModeOverlay">true</item>
```
Comment (Kaned1as) [inline src/main/res/values/styles.xml]: What was wrong with this?

File: src/main/java/com/kanedias/vanilla/plugins/saf/SafUtils.java [hunk 3]
```diff
@@ -25,7 +26,8 @@
     /**
      * Finds needed file through Document API for SAF. It's not optimized yet - you can still gain wrong URI on
      * files such as "/a/b/c.mp3" and "/b/a/c.mp3", but I consider it complete enough to be usable.
-     * @param currentDir - document file representing current dir of search
+     *
+     * @param currentDir            - document file representing current dir of search
```
Comment (Kaned1as) [inline src/main/java/com/kanedias/vanilla/plugins/saf/SafUtils.java]: Don't introduce changes if they are not necessary

**Review by Kaned1as — COMMENTED** [review]
Overall: looks good. But please test building and running plugins (of course with necessary android->x changes) with this branch. Add at least one PR to the plugins, I then will re-trigger Travis build after merging this.

**Comment by TacoTheDank** [comment]
Alright, so let's say we have these three releases, one for each of the three commits:
image

Checking Jitpack, they are now releases!
image

I clone and locally clone vanilla-music-cover-fetch. (Make sure to remember to add maven { url "https://jitpack.io" } to the root build.gradle!) I replace ```implementation project(':plugin-commons')``` with 
```implementation 'com.github.TacoTheDank:vanilla-music-plugin-commons:v1.0.0'``` (and delete the submodule folder and .gitmodules, as we no longer need those)... and there we go :D
image

**Comment by Kaned1as** [comment]
@TacoTheDank , I get it, I know how Jitpack works, don't worry. Use it myself in Tag Editor btw :)

File: src/main/res/values/styles.xml (hunk 2 above)
Comment (TacoTheDank) [reply src/main/res/values/styles.xml]: That's one of the things that are changed when migrating app to appCompat

Comment (Kaned1as) [reply src/main/res/values/styles.xml]: Okay, apparently needed for compat activity. See if you have any deviations in how activity looks like in plugins.

**Comment by TacoTheDank** [comment]
@Adonai kk, just making sure lol (explaining that was actually fun anyways), I'll make some changes to the releases (the unnecessary ones you stated), re-release the releases, and then test the library.

File: src/main/java/com/kanedias/vanilla/plugins/saf/SafUtils.java (hunk 3 above)
Comment (TacoTheDank) [reply src/main/java/com/kanedias/vanilla/plugins/saf/SafUtils.java]: gotcha

**Comment by TacoTheDank** [comment]
@Adonai Aight, tested it as a library, works perfectly as far as I can tell. Just don't know whether it will be under ```com.github.vanilla-music``` or ```com.github.Adonai``` (I hope I'm right, as I'm making a pull request to vanilla-music-cover-fetch).

Also when you make these releases, you should probably follow the 1.0.0 1.1.0 1.2.0 thing I did. Semantic versioning ftw

**Comment by Kaned1as** [comment]
Understood. Use it from vanilla-music in your PRs. I'll tag it as it gets merged.

**Comment by TacoTheDank** [comment]
@Adonai Actually, never mind about a lot of this. Using the App to AppCompat library release bloats the APKs really badly (in Cover Fetch's case, 485KB to ***over 1.4MB*** (release APKs with minifyEnabled false)), which is honestly very detrimental to storage. Attempting to use the AndroidX library release causes errors in development that cannot be fixed without prior library migration of App to AppCompat.

Therefore, this is actually going to just be a one-commit pull request that consists solely of:
- converting this into an implementable library
- simplifying a return

The first commit is already there. You can review it and decide its fate.

**Comment by TacoTheDank** [comment]
👍

**Comment by TacoTheDank** [comment]
Don't forget to tag v1.0.0 now ;)

**Comment by Kaned1as** [comment]
@TacoTheDank done!

**Pull request merged**